src/Features/Core/Portable/RelatedDocuments/AbstractRelatedDocumentsService.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Remote;
+
+namespace Microsoft.CodeAnalysis.RelatedDocuments;
+
+internal abstract class AbstractRelatedDocumentsService : IRelatedDocumentsService
+{
+    public async ValueTask GetRelatedDocumentIdsAsync(
+        Document document, int position, Func<DocumentId, CancellationToken, ValueTask> callbackAsync, CancellationToken cancellationToken)
+    {
+        var project = document.Project;
```

Comment:
core abstract impl.  does nothing but attempt ot call out to oop to do this work sicne we're about to do a lot of binding.  would be subclassed in VB as well if we want to support that.